crates/ty_test/src/lib.rs

```diff
@@ -294,6 +295,10 @@ fn run_test(
     let failures: Failures = test_files
         .into_iter()
         .filter_map(|test_file| {
+            if !KNOWN_PULL_TYPES_FAILURES.contains(&&*relative_fixture_path.as_str().replace('\\', "/")) {
+                pull_types(db, test_file.file);
+            }
```

Comment:
I tried for quite a while to use `catch_unwind` like we already do in `ty_project`:

https://github.com/astral-sh/ruff/blob/86e5a311f05943095f8e7480dea1a69fdc8c787e/crates/ty_project/tests/check.rs#L132-L151

The advantage is that if a file starts passing, we're forced to remove it from `KNOWN_PULL_TYPES_FAILURES`.

I couldn't get similar logic to work here, however, and I couldn't figure out why. Even when the `pull_types` function was panicking on a given file, the surrounding `catch_unwind` call appeared to be returning an `Ok()` variant for some reason.

Reply:
I'd have to debug this. Is there any inner or outer catch unwind?

Reply:
I think the problem here is that `KNOWN_PULL_TYPES_FAILURE` is at the mdtest file level, but `filter_map` runs for every test inside the MDTEST file (but pull types doesn't panic for all tests in that file)